### PR TITLE
Add more metrics of LRU crawler

### PR DESCRIPTION
**Pull request by f110**
Add some metrics and settings to observe a LRU crawler.

### CODE REVIEW

**Comment by f110** [comment]
Hi @grobie 

Would you please review this PR?

File: main.go [hunk 1]
```diff
@@ -173,6 +190,90 @@ func NewExporter(server string, timeout time.Duration) *Exporter {
 			nil,
 			nil,
 		),
+		lruCrawlerEnabled: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "lru_crawler", "enabled"),
```
Comment (SuperQ) [inline main.go]: Let's change `"lru_crawler"` into a `const`.

File: main.go [hunk 2]
```diff
@@ -173,6 +190,90 @@ func NewExporter(server string, timeout time.Duration) *Exporter {
...
+			nil,
+		),
+		lruCrawlerReclaimed: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "lru_crawler", "reclaimed"),
+			"Total items freed by LRU Crawler.",
+			nil,
+			nil,
+		),
+		lruCrawlerItemsChecked: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "lru_crawler", "items_checked"),
```
Comment (SuperQ) [inline main.go]: These counter values should be named `_total`.

**Comment by f110** [comment]
Thank you for review.

I fixed some issues. Could you please review again?

File: main.go [hunk 3]
```diff
@@ -482,6 +638,18 @@ func parse(stats map[string]string, key string) float64 {
...
 
+func parseBool(stats map[string]string, key string) float64 {
+	switch stats[key] {
+	case "yes":
+		return 1
+	case "no":
+		return 0
+	default:
+		log.Errorf("Failed parse %s %q", key, stats[key])
+		return math.NaN()
```
Comment (SuperQ) [inline main.go]: I'm not sure if I like this, we prefer to not return `NaN` unless there's a math reason to do so.

In this case I think we should either return 0 for default, or also return an `error` to be checked by the caller.

Comment (f110) [reply main.go]: I'm using a `NaN` for an error. it's same usage as `func parse()` , I think.

I will change `func parseBool()` to return an `error` and check by caller.
So `func parse()` should return an `error`. What do you think?

Comment (SuperQ) [reply main.go]: Yea, you're right. IMO `func parse()` is also incorrect in this behavior. But for consistency, I think this is fine. We can make a separate issue to fix the behavior separately.

**Comment by f110** [comment]
Thank you so much!

**Pull request merged**